dev/PipsPager/PipsPager.cpp

```diff
@@ -516,6 +477,93 @@ void PipsPager::OnNextButtonClicked(const IInspectable& sender, const winrt::Rou
     SelectedPageIndex(SelectedPageIndex() + 1);
 }
 
+void PipsPager::OnGotFocus(const winrt::RoutedEventArgs& args)
+{
+    if (const auto btn = args.OriginalSource().try_as<winrt::Button>())
+    {
+        // If the element inside the Pager is already keyboard focused
+        // and the user will use the mouse to focus on something else
+        // the LostFocus will not be triggered on keyboard focused element
+        // while GotFocus will be triggered on the new mouse focused element.
+        // We account for this scenario and update m_isFocused in case
+        // user will use mouse while being in keyboard focus.
+        if (btn.FocusState() != winrt::FocusState::Pointer)
+        {
+            m_isFocused = true;
+            UpdateNavigationButtonVisualStates();
+        }
+        else
+        {
+            m_isFocused = false;
+        }
+    }
+}
+
+// In order to avoid switching visibility of the navigation buttons while moving focus inside the Pager,
+// wee'll check if the next focused element is inside the Pager.
+void PipsPager::LosingFocus(const IInspectable& sender, const winrt::LosingFocusEventArgs& args)
+{
+    if (const auto repeater = m_pipsPagerRepeater.get())
+    {
+        if (const auto nextFocusElement = args.NewFocusedElement().try_as<winrt::UIElement>())
+        {
+            m_ifNextFocusElementInside = repeater.GetElementIndex(nextFocusElement) != -1
+                || nextFocusElement == m_previousPageButton.get()
+                || nextFocusElement == m_nextPageButton.get();
+        }
+    }
+}
+
+void PipsPager::OnLostFocus(const winrt::RoutedEventArgs& args)
+{
+    if (!m_ifNextFocusElementInside)
+    {
+        m_isFocused = false;
+        UpdateNavigationButtonVisualStates();
+    }
+}
+
+void PipsPager::OnPointerEntered(const winrt::PointerRoutedEventArgs& args) {
```

Comment:
Are the changes below just moving the functions a bit down in the file?

Reply:
yup, I'm just grouping all the listeners together